client/internal/wssender.go

```diff
@@ -56,13 +66,22 @@ out:
 			s.sendNextMessage()
 
 		case <-ctx.Done():
+			s.err = s.sendCloseMessage()
```

Comment:
This means stopping the sender also closes the connection which might be unexpected because run should simply run until the context is cancelled but it also closes the connection. I think sending Close message should be part of client `Stop`.

Reply:
This is by design. Stopping the sender would always try to send the close message for the close handshake and `runOneCycle` is responsible for the subsequent steps of connection closing. I updated the comment of the `runOneCycle()`.